### DISCUSSION TITLE
only validate and reValidate onSubmit

### DISCUSSION THREAD

**Discussion by dwickander** [react-hook-form/react-hook-form#5605] [Q&A] [answered]
Is there a way to only have react-hook-form validate on onSubmit? When I useForm({mode: 'onSubmit',reValidateMode: 'onSubmit',}) the initial validation happens on submit, but then it will run validation onChange after that. The documentation states that the reValidationMode of 'onSubmit' is the same as onChange. So, I am wondering if there is any way to override this?

**Comment by bluebill1049** [comment, marked as answer]
This could be a bug, can you share a codesndbox?

**Reply by dwickander** [reply]
Sorry, this was my bad. Looks like this is working as expected now. I was adding my settings to the wrong component before 😳 -- it was a Monday :-)

**Reply by bluebill1049** [reply]
👍 all good
